toxcore/MDevice.c

```diff
@@ -829,6 +853,11 @@ static int handle_packet_sync(Tox *tox, uint32_t dev_num, uint8_t *pkt, uint16_t
             break;
         }
 
+        case MDEV_SYNC_DEVICE: {
+            printf("%d\n", pkt[0]);
```

Comment:
pkt[0] is just MDEV_SYNC_DEVICE_COUNT:

you have to copy how MDEV_SYNC_FRIEND_COUNT does it